tutorial.md

```diff
@@ -1,3 +1,1555 @@
+?vd
+```
+
+
+```python
+vd.__doc__
+```
+
+
+
+
+    'This is a python script to help Translation Project Managers keep a clean overview of the vendors for a particular project by narrowing down the data in the excel.\n\n    It contains:\n        - 4 functions: \n            `ChangeDictionaryValue` to easily change values in a certain dictionary.\n            `CheckVendorMail` to easily check if an e-mail address is valid.\n            `CheckWordRate` to check if a word rate falls between EUR 0.01 and EUR 0.15 per word.\n            `CheckCatTool` to validate the CAT tool provided.\n        - 1 class, `VendorData` that has 3 class attributes: `CatTools`, the CAT Tools that can be used for the project,\n            `Keys`, the keys of the dictionary that can be modified, and `Statuses`, the possible statuses a vendor can have.\n            The class takes 4 positional arguments and 4 optional arguments:\n                1. `ProjectName` (str): the name of the translation project\n                2. `SourceLang` (str): the source language, i.e. the language the translator will translate from\n                3. `TargLang` (str): the target language, i.e. the language the translator will translate into\n                4. `VendorName` (str): the translator\'s/vendor\'s name\n                5. `VendorMail` (str): the vendor\'s e-mail address, by default empty\n                6. `WordRate` (float): the vendor\'s word rate in EUR/word, by default set to "None"\n                7. `CatTool` (str): the CAT Tool the translator will use, by default set to "XTM"\n                8. `Preferred` (bool): indicates if a vendor is a preferred vendor or not, by default set to "None"\n        - `Argparse` inside `if __name__ == "__main__":` statement to receive arguments when the script is run\n    '
+
+
+
+### Instantiating an instance of VendorData
+Now that we know the different fucntionalities we can instantiate an instance of the class VendorData.
+To illustrate the code to it's full extent, we'll instantiate a couple of them:
+
+
+```python
+VendorNL = vd.VendorData("Tutorial", "English", "Dutch", "Elmo Geeraerts") #An instance with only the positional arguments provided
+VendorFR = vd.VendorData("Tutorial", "English", "French", "Jean Lefèvre", "j.lefevre@hotmail.com", 0.10, "Trados Studio", True) #An instance with all the arguments provided
+VendorES = vd.VendorData("Tutorial", "English", "Spanish", "Emilio De La Banda", WordRate = 0.09, Preferred = False) #An instance with some of the optional arguments
+```
+
+We can check the values of the arguments for the three vendors as follows:
+
+
+```python
+VendorNL.VendorName
+```
+
+
+
+
+    'Elmo Geeraerts'
+
+
+
+
+```python
+VendorFR.VendorMail
+```
+
+
+
+
+    'j.lefevre@hotmail.com'
+
+
+
+
+```python
+VendorES.WordRate
+```
+
+
+
+
+    0.09
+
+
+
+Some arguments had default arguments. When we check the e-mail for VendorNL, for which we didn't provide one, we'll see that it does not return anything:
```

Comment:
Some arguments have default _values_.

I wouldn't say an empty string is a default value from the user's perspective. It's rather allowed to be empty.